packages/frontend/app/routes/settings.tsx

```diff
@@ -39,7 +43,7 @@ export const loader = async ({ request }: LoaderFunctionArgs) => {
 
     return { responseData }
   } else {
-    return redirect(
+    return redirectDocument(
```

Comment:
is this the only place this is needed?

Reply:
We only found issue with the settings page.

Reply:
@mkurapov  is actually a very good question. It made me realise that we need to make improvements on catching direct references to the login page and account recovery page as well! Both were already redirecting to the kratos endpoint when the `login` / `Forgot password?` buttons were clicked but weren't handling direct navigation optimally and should also be using `redirectDocument` if they are navigated to directly instead of just throwing an error.

I'm going to include an additional commit now that covers this as well.